core/store/src/lib.rs

```diff
@@ -291,28 +291,79 @@ impl NodeStorage {
     }
 }
 
+pub fn retrieve_shard_uid_from_db_key(key: &[u8]) -> io::Result<ShardUId> {
+    // TODO(reshardingV3) Consider changing the Error type to `StorageError`?
+    // Would need changing error types for `Store` methods as well.
+    ShardUId::try_from(&key[..8])
+        .map_err(|e| io::Error::other(format!("Could not retrieve ShardUId from db key: {}", e)))
+}
+
+pub fn replace_shard_uid_key_prefix(key: &[u8], shard_uid: ShardUId) -> [u8; 40] {
```

Comment:
Add a comment please.

It seems a bit sus performance wise - it copies the entire key. Would it make sense to overwrite the prefix in place on a mutable key? 

nit: I would be tempted to make the mapped shard uid a newtype to enforce compiler to warn us if the non-mapped shard uid is every passed in here.

Reply:
The method was removed in next commit.
I considered making `key` mut and overwriting the prefix, but `key` comes from a very high level in the stake and might require a huge refactor.